src/nvim/message.c

```diff
@@ -2383,6 +2383,15 @@ static void redir_write(char_u *str, int maxlen)
   char_u      *s = str;
   static int cur_col = 0;
 
+  // Append output to capture().
+  if (capture_ga) {
+    int len = maxlen > 0 ? maxlen : (int)STRLEN(str);
+    ga_grow(capture_ga, len);
+    memmove((char *)capture_ga->ga_data
```

Comment:
Actually not `memmove` and not `memcpy` and not `ga_grow`. Neovim has `ga_concat_len` function just for this purpose. Basically

```C
if (len < 0) {
  ga_concat(capture_ga, str);
} else {
  ga_concat_len(capture_ga, str, len);
}
```

if you accept the renaming of `maxlen`. Or

```C
size_t len = STRLEN(str);
if (maxlen < 0 && len > maxlen) {
  len = (size_t) maxlen;
}
ga_concat_len(capture_ga, str, len);
```

if you do not. I do not know why Vim does not have this function, I added it in a somewhat similar situation.